process/block/metrics.go

```diff
@@ -76,6 +78,45 @@ func getMetricsFromHeader(
 	appStatusHandler.SetUInt64Value(core.MetricNumProcessedTxs, uint64(totalTx))
 }
 
+func saveMetricsForACommittedBlock(
+	appStatusHandler core.AppStatusHandler,
+	isInConsensus bool,
+	currentBlockHash string,
+	highestFinalBlockNonce uint64,
+	headerMetaNonce uint64,
+) {
+	if isInConsensus {
+		appStatusHandler.Increment(core.MetricCountConsensusAcceptedBlocks)
+	}
+	appStatusHandler.SetStringValue(core.MetricCurrentBlockHash, currentBlockHash)
+	appStatusHandler.SetUInt64Value(core.MetricHighestFinalBlockInShard, highestFinalBlockNonce)
+	appStatusHandler.SetStringValue(core.MetricCrossCheckBlockHeight, fmt.Sprintf("meta %d", headerMetaNonce))
+}
+
+func estimateRewardsForMetachain(
```

Comment:
why do you need to reestimate ? is it not already calculated ?

Reply:
This method has a wrong name, because this method actually check if node that is in meta chain signed for a  block and count how many headers was in block .  Name of method was changed.